Failed to build the cmake(3.23.1) on Centos7

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.


### Version

4f9f01c785cfea7ae4cc66f9c6cf55fa0c4866d9

### Minimal reproduce step

```
$ mkdir build
$ sh build.sh build
```

### What did you expect to see?

Can build the newer cmake version

### What did you see instead?

```
loading initial cache file /incubator-kvrocks/build/cmake-3.23.1/Bootstrap.cmk/InitialCacheFlags.cmake
-- Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY OPENSSL_INCLUDE_DIR)
CMake Error at Utilities/cmcurl/CMakeLists.txt:586 (message):
  Could not find OpenSSL.  Install an OpenSSL development package or
  configure CMake with -DCMAKE_USE_OPENSSL=OFF to build without OpenSSL.
```

### Anything Else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

--- a/build.sh
+++ b/build.sh
@@ -60,7 +60,7 @@ if [ "$(printf '%s\n' "$CMAKE_REQUIRE_VERSION" "$CMAKE_VERSION" | sort -V | head
     CMAKE_DOWNLOAD_VERSION=3.23.1
     curl -O -L https://github.com/Kitware/CMake/releases/download/v$CMAKE_DOWNLOAD_VERSION/cmake-$CMAKE_DOWNLOAD_VERSION.tar.gz
     tar -zxf cmake-$CMAKE_DOWNLOAD_VERSION.tar.gz && cd cmake-$CMAKE_DOWNLOAD_VERSION
-    ./bootstrap --prefix=$CMAKE_INSTALL_DIR && make && make install && cd ../..
+    ./bootstrap --prefix=$CMAKE_INSTALL_DIR -- -DCMAKE_USE_OPENSSL=OFF && make && make install && cd ../..
     CMAKE_BIN=$CMAKE_INSTALL_DIR/bin/cmake
 fi
 

